### PR TITLE
Fix container image state and add warning for signatures

### PR DESCRIPTION
**Pull request by saschagrunert**
Fixing the feature state and adding a warning about the non matching signatures issue.

cc @kubernetes/release-engineering 

Preview: https://deploy-preview-44062--kubernetes-io-main-staging.netlify.app/releases/download/#container-images

### CODE REVIEW

File: content/en/releases/download.md [hunk 1]
```diff
@@ -33,20 +33,28 @@ Find your preferred operating system below.
...
 
+{{< warning >}}
+
+{{< feature-state for_k8s_version="v1.26" state="beta" >}}
+
+All sigstore signatures do currently not match across different geolocations.
+More information about this problem is available in the [corresponding GitHub
+issue](https://github.com/kubernetes/registry.k8s.io/issues/187).
+
+{{< /warning >}}
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply content/en/releases/download.md]: Thanks! I applied the suggestion.

File: content/en/releases/download.md [hunk 2]
```diff
@@ -33,20 +33,35 @@ Find your preferred operating system below.
 All Kubernetes container images are deployed to the
 `registry.k8s.io` container image registry.
 
-{{< feature-state for_k8s_version="v1.24" state="alpha" >}}
-
 For Kubernetes {{< param "version" >}}, the following
-container images are signed using [cosign](https://github.com/sigstore/cosign)
+container images are signed using [sigstore](https://sigstore.dev)
 signatures:
```
Comment (sftim) [inline content/en/releases/download.md:38]:
```suggestion
```

File: content/en/releases/download.md
Comment (sftim) [inline content/en/releases/download.md:1]: The existing text:
```markdown
The Kubernetes project publishes a list of signed Kubernetes container images
in [SPDX 2.3](https://spdx.dev/specifications/) format.
You can fetch that list using:
```shell
curl -Ls "https://sbom.k8s.io/$(curl -Ls https://dl.k8s.io/release/stable.txt)/release" | grep "SPDXID: SPDXRef-Package-registry.k8s.io" |  grep -v sha256 | cut -d- -f3- | sed 's/-/\//' | sed 's/-v1/:v1/'
```
For Kubernetes v{{< skew currentVersion >}}, the only kind of code artifact that
you can verify integrity for is a container image, using the experimental
signing support.
To manually verify signed container images of Kubernetes core components, refer to
[Verify Signed Container Images](/docs/tasks/administer-cluster/verify-signed-artifacts).
```

could move into the “Container image signatures” section. However, the information about signature verification might also partly wrong, now we have official packages.

**Comment by sftim** [comment]
Once this gets LGTM, let's merge it.

/approve

**Pull request merged**